Error when trying to install

I am running a Vue 3 project on storybook.js 

I tried to install using the following but am receiving an error ERESOLVE unable to resolve dependency tree

```
npm i --save-dev @whitespace/storybook-addon-html prettier react-syntax-highlighter
```

I ended up running this command to get it to install 
```
npm i --save-dev @whitespace/storybook-addon-html prettier react-syntax-highlighter --legacy-peer-deps 
```

now when i try and run storybook i'm getting another error 
Error: Cannot find module 'react/package.json'

Just wondering if this is something on my end and if anyone has any ideas on how i can fix it. 

Thanks.

node v16.15.1
npm 8.15.0

--- a/package.json
+++ b/package.json
@@ -78,7 +78,7 @@
     "@storybook/core-events": "^6.5.8",
     "@storybook/theming": "^6.5.8",
     "prettier": "^2.0.0",
-    "react": "^16.8.0 || ^17.0.0",
+    "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
     "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0",
     "react-syntax-highlighter": "^15.0.0"
   },

